### DISCUSSION TITLE
Why Prisma `findFirst` Relation Return `null`?

### DISCUSSION THREAD

**Discussion by kcoet** [prisma/prisma#15685] [Q&A] [answered]
Why does prism return `null` when I use `findFirst` in the relation? 

I have tried using Postgresql and it works fine, but not in Mongodb.

I didn't find any documentation that `findFirst` doesn't work in Mongodb, so I'm assuming it should. https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#findfirst

This is my schema:

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model Game {
  id                          String                        @id @default(auto()) @map("_id") @db.ObjectId
  name                        String

  level                       Level                          @relation(fields: [levelId], references: [id])
  levelId                     String                        
}

model Level {
  id                          String                        @id @default(auto()) @map("_id") @db.ObjectId
  name                        String                        @unique

  games                       Game[]
}
```

And this is the code I mean:

```js
const gameForBeginner = await prisma.game.findFirst({
    where: {
      level: {
        name: "beginner",
      },
    },
  });

console.log(gameForBeginner); // null
```

Though I have a lot of games:

```js
const games = await prisma.game.findMany({
  include: {
    level: true,
  },
});

console.log(games);
// [
//   {
//     id: '633e96bfe022c9818ca1f267',
//     name: 'discrete',
//     levelId: '633e96bee022c9818ca1f264',
//     level: { id: '633e96bee022c9818ca1f264', name: 'beginner' }
//   },
//   {
//     id: '633e96bfe022c9818ca1f268',
//     name: 'blind',
//     levelId: '633e96bee022c9818ca1f264',
//     level: { id: '633e96bee022c9818ca1f264', name: 'beginner' }
//   },
//   {
//     id: '633e96bfe022c9818ca1f269',
//     name: 'detailed',
//     levelId: '633e96bee022c9818ca1f264',
//     level: { id: '633e96bee022c9818ca1f264', name: 'beginner' }
//   },
//   {
//     id: '633e96bfe022c9818ca1f26a',
//     name: 'villainous',
//     levelId: '633e96bee022c9818ca1f264',
//     level: { id: '633e96bee022c9818ca1f264', name: 'beginner' }
//   },
//   {
//     id: '633e96bfe022c9818ca1f26b',
//     name: 'lawful',
//     levelId: '633e96bee022c9818ca1f264',
//     level: { id: '633e96bee022c9818ca1f264', name: 'beginner' }
//   }
// ]
```

**Comment by austincrim** [comment, marked as answer]
Hey @kcoet 👋,

I was able to recreate this on my end as well. Would you mind opening a [bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) with all the relevant details?

**Reply by kcoet** [reply]
Hi @austincrim , sure, here it is: https://github.com/prisma/prisma/issues/15702

**Comment by Mnigos** [comment]
I'm also facing similar issue. 

```ts
    const lastHistoryTrack = await this.prisma.historyTrack.findFirst({
      where: {
        user: {
          id,
        },
      },
      orderBy: {
        playedAt: 'desc',
      },
      select: {
        playedAt: true,
        spotifyTrackId: true,
        user: {
          select: {
            id: true,
          },
        },
      },
    })
```
    
This sometimes resolves with correct result and sometimes null. I don't know why this happens.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Reply by Mnigos** [reply]
I don't think this issue has been resolved. I'm receiving null when i repeat my query every minut, then every third response returns null instead of actual value.
